Spooky Spectral Instruments

Mercenaryblue opened the pull request:
:cl: 
Add: Skeletons can now have their own spectral instruments
Add: \[Dooting Intensifies\]
admin: the variable "too_spooky" defines if it will spawn new instruments, by default TRUE.
/:cl:

## **Things are about to get Spooky**
Imagine the scariest thing you can.
No, not that old god rising from the sacrifice of the blood cult.
No, not the singularity eating what little safe space you had.
No, not the artifical intelligence deciding to purge all organic lifeforms.

**S K E L E T O N S.**
You have one inside you, right now.
Its waiting the right moment to get out.
To rip apart its fleshy prison.
And play some mournful jazz.
To bring about the spooky-pocalypse.

And the instrument of your doom is a trumpet.
Or a trombone, really.
Even a saxophone could do.
And they'll play it right into your soul.

<a href="https://imgur.com/Raa7fzo"><img src="https://i.imgur.com/Raa7fzo.png" title="source: imgur.com" /></a>
Happy Spookyween


**What is a spectral trumpet?**
Its an instrument only a skeleton can use. When attacked by it, you receive stamina damage until you give up your mortal life, at which point you will rise as a new skeleton. Otherwise, it acts the same as any other instrument. Also comes with saxophones and trombone variety.

placeholder trumpet sprite: <a href="https://imgur.com/BTb7PMu"><img src="https://i.imgur.com/BTb7PMu.png" title="source: imgur.com" /></a>
_p.s: trumpets currently use the trombone in-hands and instrument sounds. Just play in high notes._

Mercenaryblue committed:
[dooting intensifies]

Spectral Instruments added
To be used with Spooky Skeletons.

Mercenaryblue commented:
That IconDiffBot is  👍 👍 👍

AnonymousNow commented:
Isn't the trinity of skeleton instruments the trombone, the xylophone and the bongos? Are we getting those?

RemieRichards commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -110,20 +110,54 @@
 	instrumentId = "trombone"
 
+/obj/item/device/instrument/brass/trombone/spectral
+	name = "spectral trombone"
```
trom BONE

Thunder12345 replied:
2spooky

DoctoreMad commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -164,3 +198,60 @@
+			H.set_species(/datum/species/skeleton)
+			H.visible_message("<span class='warning'>[H] has given up on life as a mortal.</span>")
+			to_chat(H, "<B>You are the spooky skeleton!</B>")
+			to_chat(H, "A new life and identity has begun. Help your fellow skeletons into bringing out the spooky-pocalypse. You haven't forgotten your past life, but you are no longer beholden to any past loyalties.")
```
I believe you need admin ruling to be able to say "Past allegiances are foregone" in this text unless there's something I don't know going about.

Mercenaryblue replied:
They can only be spawned by admins anyway.

kevinz000 commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -164,3 +198,60 @@
+			return ..() //undead are unaffected by the spook-pocalypse.
+		if(istype(H.dna.species, /datum/species/zombie))
+			H.adjustStaminaLoss(25) //zombies can't resist the doot
+		C.Knockdown(15)
```
One hit stunlock convert that isn't counter able is shit.

Mercenaryblue replied:
I could alter it to be simply raw stamina loss.

Mercenaryblue replied:
Done. They will no longer stun living creatures.

Mercenaryblue committed:
Spectral Re-balance

The spectral instruments no longer stun living creatures.
Text make you retain your loyalties now.

ShizCalev commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -164,3 +198,58 @@
+	var/t = stripped_input(H, "Enter your new skeleton name", H.real_name, null, MAX_NAME_LEN)
+	if(!t)
+		t = "spooky skeleton"
+	H.real_name = t
```
H.fully_replace_character_name(H.real_name, t)

DoctoreMad commented:
Make double-haunted version with the same procs except when a person is converted they also get a double-haunted instrument.

Mercenaryblue committed:
Requested Changes

Added variable "too_spooky".
Defines if it will spawn a new instrument.
Re-balanced damage to 5 hits average per conversion.
Added admin notice to changelog.

DoctoreMad commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -164,3 +202,68 @@
+				else
+					to_chat(H, "The spooky gods forgot to ship your instrument. Better luck next unlife.")
+			to_chat(H, "<B>You are the spooky skeleton!</B>")
+			to_chat(H, "A new life and identity has begun. Help your fellow skeletons into bringing out the spooky-pocalypse. You haven't forgotten your past life, but you are no longer beholden to any past loyalties.")
```
What happened? You reverted it to the old text. I assume a admin/coder told you that it didn't matter?

Mercenaryblue replied:
Wow. I was certain I changed that. I guess I may have reverted it by accident, or forgot to copy-paste that part over.

Mercenaryblue committed:
Loyalty text fix

DoctoreMad commented:
For reference, it reverted on your last git push, "Requested changes"

Mercenaryblue commented:
It was probably my sticky middle mouse at fault.
The requested changes are all there. The PR is ready.

DoctoreMad commented:
ShizCalev's requested change still shows up as being requested even though the code change is done
What gives

Mercenaryblue commented:
I think he has to change it manually.

AnturK commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -110,20 +111,57 @@
+	item_state = "saxophone"
+	instrumentId = "saxophone"
+	force = 0
+	attack_verb = list("played","jazzed","saxxed","mourned","dooted")
```
Dunno if it's intended but you didn't set the spooky here

AnturK commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -9,6 +9,7 @@
 	var/datum/song/handheld/song
 	var/instrumentId = "generic"
 	var/instrumentExt = "mid"
+	var/too_spooky = FALSE //this define if the instrument will spawn a copy upon dooting a new skeleton.
```
This one is bit ugly. Do instruments even have any unique properties besides sound id and visuals so it can't be spectral/trumpet, spectral/saxophone etc ?

Mercenaryblue replied:
They all behave the same, the only difference is which sound board they use.

AnturK replied:
Could you make the spectralness a component please ? It just needs to do things on on hit after all, and wouldn't need any weird pathing or special variables  on base types that way. (And could be applied to anything for admin abuse)

Mercenaryblue replied:
I'll do my best.

Mercenaryblue committed:
Spooky Trumpets

Removed comment.
Added missing spookiness.

DoctoreMad commented on code/game/objects/items/devices/instruments.dm:
```diff
@@ -143,7 +143,7 @@
 	force = 0
 	attack_verb = list("played","jazzed","saxxed","mourned","dooted")
-
+	too_spooky = TRUE
```
Why did you add this here if you didn't remove the flag in the base /instrument item

Mercenaryblue replied:
To allow it to spawn new instruments.

DoctoreMad commented:
@ShizCalev Confirm this change reeeee

Mercenaryblue committed:
Spooky Dooting

Major Overhaul.

removed /brass/
Spooky Dooting is now a component.
Boneless humanoids no longer lose stamina.

Mercenaryblue committed:
Spooky Dooting

mk2
